cdap-support-bundle/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -111,4 +139,109 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
+    File baseDirectory = new File(cConf.get(Constants.SupportBundle.LOCAL_DATA_DIR));
+    if (!baseDirectory.exists()) {
+      LOG.debug("No content in Support Bundle.");
+      return;
+    }
+    List<SupportBundleOperationStatus> bundleOperationStatusList = new ArrayList<>();
+    DirUtils.listFiles(baseDirectory)
+      .stream()
+      .filter(file -> !file.isHidden() && file.isDirectory())
+      .forEach(uuidFile -> {
+        SupportBundlePipelineStatus bundlePipelineStatus = new SupportBundlePipelineStatus();
+        SupportBundleOperationStatus bundleOperationStatus =
+          new SupportBundleOperationStatus(uuidFile.getName(), CollectionState.INVALID, bundlePipelineStatus);
+        try {
+          bundleOperationStatus = bundleGenerator.getBundle(uuidFile.getName());
+        } catch (IOException e) {
+          LOG.debug("Can not find status json: {}", uuidFile.getName());
+        }
+        bundleOperationStatusList.add(bundleOperationStatus);
+      });
+    responder.sendString(HttpResponseStatus.OK, GSON.toJson(bundleOperationStatusList));
+  }
+
+  /**
+   * Get the specific bundle status from status.json by using uuid.
+   *
+   * @param uuid the bundle id which is also the uuid
+   * @throws NotFoundException is thrown if status.json is not found
+   */
+  @GET
+  @Path("/support/bundles/{uuid}/status")
+  public void getSupportBundleStatus(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    SupportBundleEntityId bundleEntityId = new SupportBundleEntityId(uuid);
+    contextAccessEnforcer.enforce(bundleEntityId, StandardPermission.GET);
+    SupportBundleOperationStatus bundleOperationStatus = bundleGenerator.getBundle(uuid);
+    responder.sendString(HttpResponseStatus.OK, GSON.toJson(bundleOperationStatus));
+  }
+
+  /**
+   * Delete the specific bundle by using the uuid
+   *
+   * @param uuid the bundle id which is also the uuid
+   * @throws NotFoundException is thrown if status.json is not found
+   */
+  @DELETE
+  @Path("/support/bundles/{uuid}")
+  public void deleteSupportBundle(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    SupportBundleEntityId bundleEntityId = new SupportBundleEntityId(uuid);
+    contextAccessEnforcer.enforce(bundleEntityId, StandardPermission.DELETE);
+    File uuidFile = bundleGenerator.getUUIDFile(uuid);
+    if (!uuidFile.exists()) {
+      throw new NotFoundException(String.format("No such uuid '%s' in Support Bundle.", uuid));
+    }
+    bundleGenerator.deleteBundle(uuidFile);
+    responder.sendString(HttpResponseStatus.OK, String.format("Successfully deleted bundle %s", uuid));
+  }
+
+  /**
+   * Exports all selected files as a ZIP archive file.
+   *
+   * @param uuid the bundle id which is also the uuid fileJson body
+   */
+  @POST
+  @Path("/support/bundle/{uuid}/download")
```

Comment:
I prefer just using the `/support/bundles/{uuid}` to fetch the full bundle (also missing `s` in `bundles`).